axum/src/extract/content_length_limit.rs

```diff
@@ -38,20 +42,29 @@ where
     type Rejection = ContentLengthLimitRejection<T::Rejection>;
 
     async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
-        let content_length = req.headers().get(http::header::CONTENT_LENGTH);
-
-        let content_length =
-            content_length.and_then(|value| value.to_str().ok()?.parse::<u64>().ok());
-
-        if let Some(length) = content_length {
-            if length > N {
+        let content_length = req
+            .headers()
+            .get(http::header::CONTENT_LENGTH)
+            .and_then(|value| value.to_str().ok()?.parse::<u64>().ok());
+
+        match (content_length, req.method().clone()) {
+            (content_length, Method::GET | Method::HEAD | Method::OPTIONS) => {
```

Comment:
Pattern-matching fun to avoid the `clone()`.

```suggestion
        match (content_length, req.method()) {
            (content_length, &(Method::GET | Method::HEAD | Method::OPTIONS)) => {
```